### PR TITLE
test(resize): handle resize test case

### PR DESCRIPTION
**Pull request by mittachaitu**
Signed-off-by: mittachaitu <sai.chaithanya@mayadata.io>

This PR fixes the resize test case by forming data connection. 

**NOTE:**
There is dependency PR #https://github.com/openebs/libcstor/pull/25 on libcstor as a part of resize feature.

### CODE REVIEW

File: tests/cstor/gtest/test_zrepl_prot.cc [hunk 1]
```diff
@@ -1987,7 +1993,8 @@ TEST(ZvolResizeTest, ResizeZvol) {
 	ASSERT_EQ(hdr_in.len, 0);
 
 	// get the zvol size after
-	str = execCmd("zfs", std::string("get -Hpo value volsize ") + zvolname);
+	str = execCmd("zfs", std::string("get -Hpo value volsize ") +
+			zvolname + "_rebuild_clone");
```
Comment (mynktl) [inline tests/cstor/gtest/test_zrepl_prot.cc:1997]: Can you also add a verification of resize once volume become healthy?

Comment (mittachaitu) [reply tests/cstor/gtest/test_zrepl_prot.cc:1997]: Done

File: tests/cstor/gtest/test_zrepl_prot.cc [hunk 2]
```diff
@@ -1939,13 +1939,14 @@ TEST(ZvolResizeTest, DataConn) {
...
+	SocketFd datasock;
 	zvol_io_hdr_t hdr_in, hdr_out = {0};
 	Zrepl zrepl;
 	Target target;
 	int rc, control_fd;
 	std::string host;
 	std::string str;
-	uint64_t val1, val2;
+	uint64_t val1, val2, val3, size, ioseq;
 	uint16_t port;
```
Comment (mynktl) [inline tests/cstor/gtest/test_zrepl_prot.cc:1950]: let's rename this variable.

Comment (mittachaitu) [reply tests/cstor/gtest/test_zrepl_prot.cc:1950]: Done

**Review by mynktl — APPROVED** [review]

**Pull request merged**